pyiceberg/io/pyarrow.py

```diff
@@ -2442,7 +2448,9 @@ def write_parquet(task: WriteTask) -> DataFile:
         )
         fo = io.new_output(file_path)
         with fo.create(overwrite=True) as fos:
-            with pq.ParquetWriter(fos, schema=arrow_table.schema, **parquet_writer_kwargs) as writer:
+            with pq.ParquetWriter(
+                fos, schema=arrow_table.schema, store_decimal_as_integer=True, **parquet_writer_kwargs
```

Comment:
"""
By default, this is DISABLED and all decimal types annotate fixed_len_byte_array. When enabled, the writer will use the following physical types to store decimals: 
- int32: for 1 <= precision <= 9. 
- int64: for 10 <= precision <= 18. 
- fixed_len_byte_array: for precision > 18.
"""

from https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetWriter.html
![Screenshot 2025-05-09 at 8 35 41 AM](https://github.com/user-attachments/assets/94bdeb58-6e56-4ccb-b5a4-cbff97fbe500)

Reply:
this matches the parquet data type mapping for decimal
https://iceberg.apache.org/spec/#parquet

![Screenshot 2025-05-09 at 8 36 40 AM](https://github.com/user-attachments/assets/c35ed15b-db1c-405d-ad16-0a0c1a387d87)